### ISSUE TITLE
Fluentd is restarting when handling big file through fluent-bit forward input

### ISSUE THREAD

**Issue by TomekLacina** [fluent/fluentd#1557]
FluentD 0.12
FluentBit 0.11
CentOS

FluentD Configuration
```
<source>
  @type  http
  @id    input1
  @label @mainstream
  format /^(?<customer>\w*);(?<year>(\d{4}))-(?<month>(\d{2}))-(?<day>(\d{2})) (?<hours>(\d{2})):(?<minutes>(\d{2})):(?<sec>(\d{2})),(?<nano>(\d{3}));(?<professional>.*);(?<citizen>\w*);(?<action>\w*);(?<activity>[\w ]*);(?<activity_id>[\w ]*);"(?<activity_desc>.*)";?(?<channel>\w*)$/
  port  24224
  @log_level trace
</source>

<source>
  type forward
  bind 0.0.0.0
  port 12225
  @label @mainstream
</source>

<label @mainstream>

        <filter **>
          @type parser
          format /^(?<customer>\w*);(?<year>(\d{4}))-(?<month>(\d{2}))-(?<day>(\d{2})) (?<hours>(\d{2})):(?<minutes>(\d{2})):(?<sec>(\d{2})),(?<nano>(\d{3}));(?<professional>.*);(?<citizen>\w*);(?<action>\w*);(?<activity>[\w ]*);(?<activity_id>[\w ]*);"(?<activity_desc>.*)";?(?<channel>\w*)$/ 
          key_name log
          reserve_data true
          suppress_parse_error_log false
        </filter>
      
		<filter **>
          @type record_transformer
          <record>
            customer_key ${record["customer"]}-${record["year"]}-${record["month"]}
			@timestamp ${record["year"]}-${record["month"]}-${record["day"]}T${record["hours"]}:${record["minutes"]}:${record["sec"]}
          </record>	
        </filter>
	<match **>
	  @type elasticsearch
	  logstash_format true
	  
	  host "my host" 
	  time_key @timestamp
	  time_key_exclude_timestamp false	  	  
	  port 9200 
	  user something
	  password something_else
	  target_index_key customer_key
	  index_name fluentd 
	  type_name fluentd	 
	  
	  buffer_type file
	  buffer_path /opt/fluentd/buffer/fluentd.*.buffer
	  
	</match>
</label>
```

Fluent-bit  configuration
```
[SERVICE]
    Flush     1
    Daemon    On
    Log_Level debug
    Log_File  /data/preprod/test/logviewer2/log-recovery.log
    HTTP_Monitor on
    HTTP_Port 2020

[INPUT]
    Name tail
    Path /data/preprod/test/logviewer2/failtest.txt

[OUTPUT]
    Name forward
    Host host/with/fluentD
    Port 12225

```

 logs generated using java method
```
@Test
	public void testCostam() throws IOException {


		BufferedWriter bw = null;
		FileWriter fw = null;

		try {

			fw = new FileWriter("c:/klucze/failtest.txt");
			bw = new BufferedWriter(fw);
			for(int i = 0 ; i < 1000000; i++){
				String msg = "trinity;2017-04-15 00:00:00,180;AVS;;BULKREAD;Message;;\"\";fail_test"+i + "\n";
				bw.write(msg);
			}



			System.out.println("Done");

		} catch (IOException e) {

			e.printStackTrace();

		} finally {

			try {

				if (bw != null)
					bw.close();

				if (fw != null)
					fw.close();

			} catch (IOException ex) {

				ex.printStackTrace();

			}

		}
	}

```

I am trying to find out if log forwarding is reliable, as I have requirement that i can't loose any log while i send it from fluent-bit to fluentd which is storing them in elasticsearch. Everything works pretty well but, when i try to use in_tail plugin with fluentbit to transfer logs from single file (70mb, file generated by above function) i get errors, fluentD restarts itself few times, and in the process i am loosing around 50% of logs.

some logs frome fluent bit 
`[2017/05/04 10:17:36] [debug] [out_forward] request 2052648 bytes to flush
[2017/05/04 10:17:36] [debug] [out_fw] 23868 entries tag='tail.0' tag_len=6
[2017/05/04 10:17:36] [error] [out_fw] error writing content body
[2017/05/04 10:17:36] [error] [out_fw] error writing content body
[2017/05/04 10:17:36] [debug] [retry] re-using retry for task_id=18 attemps=2
[2017/05/04 10:17:36] [debug] [sched] retry=0x1a87650 18 in 16 seconds
[2017/05/04 10:17:36] [debug] [retry] re-using retry for task_id=26 attemps=2
[2017/05/04 10:17:36] [debug] [sched] retry=0x1aaefa0 26 in 19 seconds
[2017/05/04 10:17:36] [error] [out_fw] error writing content body
[2017/05/04 10:17:36] [debug] [retry] re-using retry for task_id=29 attemps=2
[2017/05/04 10:17:36] [debug] [sched] retry=0x1aaeef0 29 in 7 seconds
[2017/05/04 10:17:36] [error] [out_fw] error writing content body
[2017/05/04 10:17:36] [debug] [task] task_id=6 reached retry-attemps limit 2/1
[2017/05/04 10:17:36] [ warn] [engine] Task cannot be retried: task_id=6 thread_id=2 output=forward.0
[2017/05/04 10:17:36] [debug] [task] destroy task=0x1a97bc0 (task_id=6)
[2017/05/04 10:17:36] [debug] [dyntag tail.0] 0x1a96600 destroy (tag=tail.0)
[2017/05/04 10:17:36] [error] [out_fw] error writing content body
[2017/05/04 10:17:36] [debug] [retry] re-using retry for task_id=8 attemps=2
[2017/05/04 10:17:36] [debug] [sched] retry=0x1a87ab0 8 in 3 seconds`

and some logs from fluentD
`2017-05-04 08:18:19 +0000 [info]: process finished code=9
2017-05-04 08:18:19 +0000 [error]: fluentd main process died unexpectedly. restarting.
2017-05-04 08:18:19 +0000 [info]: starting fluentd-0.12.35
`

**Comment by edsiper** [comment]
Are you using Fluent Bit v0.11.4?

**Comment by TomekLacina** [comment]
Yes, i am using FluentBit v0.11.4 ,  was trying with similar setup on fluentBit v.010, but fluentd had reset more frequently

**Comment by edsiper** [comment]
Just to isolate the problem, what about if you add this option to your input section:

```
[INPUT]
    Name and
    Mem_Buf_Limit 10MB
```

Please try mem_buf_limit and let me know if that helps

**Comment by TomekLacina** [comment]
Setting Mem_Buf_Limit at 10MB at fluent-bit didn't help. FluentD is still restarting and fluent-bit. 
I also tried removing record transformers from fluentD config file, but it didn't help.

**Comment by repeatedly** [comment]
@edsiper Do you reproduce this problem on your side?

**Comment by edsiper** [comment]
@repeatedly cannot reproduce here.

@TomekLacina would you please provide a sample of your _/data/preprod/test/logviewer2/failtest.txt_ file that helps to reproduce the problem ?

**Comment by TomekLacina** [comment]
this is sample line from test data, more can be easily produced by java snipped I have posted before.

`test;2017-04-15 00:01:00,131;AVS;;BULKREAD;Message;;"";failTest0`

I have tried tweaking  configuration a bit, and it started to working. I have realised that, when I set
buffer_chunk_limit higher than 8m, fluentD is restarting, I have also switched to td-agent and fluentd provided there, and it seems to work for now.
